python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -273,52 +273,45 @@ def quantize(self, model,  # remove the type requirement for type checking
         :return:            A accelerated Pytorch-Lightning Model if quantization is sucessful.
         """
         if not accelerator or accelerator == 'onnxruntime':
-            # check if dataloader is of legal format
-            check_pytorch_dataloaders(model, [calib_dataloader, val_dataloader])
-
-            if approach not in ['static', 'dynamic']:
-                invalidInputError(False,
-                                  "Approach should be 'static' or 'dynamic', "
-                                  "{} is invalid.".format(approach))
-            approach_map = {
-                'static': 'post_training_static_quant',
-                'dynamic': 'post_training_dynamic_quant'
+            method_map = {
+                None: {
+                    'fx': 'pytorch_fx',
+                    'eager': 'pytorch',
+                    'ipex': 'pytorch_ipex',
+                    None: 'pytorch_fx'  # default
+                },
+                'onnxruntime': {
+                    'qlinear': 'onnxrt_qlinearops',
+                    'integer': 'onnxrt_integerops',
+                    None: 'onnxrt_qlinearops'   # default
+                }
             }
-            approach = approach_map.get(approach)
-            if accelerator is None:
-                framework = 'pytorch_{}'.format(method)
-            if accelerator == 'onnxruntime':
-                framework = "{}_{}ops".format('onnxrt', method)
-            quantizer = QuantizationINC(framework=framework, conf=conf, approach=approach,
-                                        tuning_strategy=tuning_strategy,
-                                        accuracy_criterion=accuracy_criterion,
-                                        timeout=timeout, max_trials=max_trials)
+            framework = method_map[accelerator].get(method, None)
```

Comment:
does it make sense to have `framework=None`?

Reply:
Yes, if `framework=None`, an error will be raised.